### PR TITLE
Improves Powershell scripts for dependency install

### LINKED ISSUE #237: setup.ps1 upgrade not working as expected
## Report

- [x] I have searched [existing issues](https://github.com/CatalystCode/3dtoolkit/issues) and this is not a duplicate.

## Environment

- OS Version: `Windows 10`

- SDK Version: `14393`

- Visual Studio Version: `17 (v141)`

## Issues and Steps to Reproduce
***Replaces this with steps to repro your issue, if applicable.***

```
git clone https://github.com/catalystcode/3dtoolkit 3dtoolkit-issue-repro
cd 3dtoolkit-issue-repro
git checkout 6887c767cc51fa2eaa3337e3fd536e2ea08ba7ce
.\setup.cmd
git checkout master
.\setup.cmd
```

## Expected Behavior
***Describe what you expected would happen.***

Install all library updates

## Actual Behavior
***Describe what actually happened. Include screenshots, if applicable.***

```
C:\Users\begreen\git\3dtoolkit
λ .\setup.cmd
Removing outdated lib
Downloading Nvpipe lib archive
Downloaded Nvpipe lib archive
Extracting...
Error retrieving Nvpipe libraries: Exception calling "ExtractToDirectory" with "2" argument(s): "The file 'C:\Users\begreen\git\3dtoolkit\Libraries\Nvpipe\Resources\paris_qcif.yuv' already exists."
Finished Library  1/5
Downloading m62patch_nvpipe_headers0x8D5C71C27D7CA16.zip from https://3dtoolkitstorage.blob.core.windows.net/libs/m62patch_nvpipe_headers.zip
Downloaded m62patch_nvpipe_headers lib archive
Clearing existing \headers
Extracting...
Finished
Downloading m62patch_nvpipe_x640x8D5CBD1343D942E.zip from https://3dtoolkitstorage.blob.core.windows.net/libs/m62patch_nvpipe_x64.zip
Downloaded m62patch_nvpipe_x64 lib archive
Clearing existing \x64
Extracting...
Finished
Downloading m62patch_nvpipe_Win320x8D5CBD12637FCBD.zip from https://3dtoolkitstorage.blob.core.windows.net/libs/m62patch_nvpipe_Win32.zip
Downloaded m62patch_nvpipe_Win32 lib archive
Clearing existing \Win32
Extracting...
Finished
Error retrieving WebRTC libraries: Exception calling "ExtractToDirectory" with "2" argument(s): "The file 'C:\Users\begreen\git\3dtoolkit\Libraries\Nvpipe\Resources\paris_qcif.yuv' already exists."
Finished Library  2/5
Error retrieving DirectXTK libraries for native client: Exception calling "ExtractToDirectory" with "2" argument(s): "The file 'C:\Users\begreen\git\3dtoolkit\Libraries\Nvpipe\Resources\paris_qcif.yuv' already exists."
Finished Library  3/5
Error retrieving WebRTC-UWP & LibYUV libraries for directx hololens client: Exception calling "ExtractToDirectory" with "2" argument(s): "The file 'C:\Users\begreen\git\3dtoolkit\Libraries\Nvpipe\Resources\paris_qcif.yuv' already exists."
Finished Library  4/5
Error retrieving OpenGL libraries for OpenGL sample server: Exception calling "ExtractToDirectory" with "2" argument(s): "The file 'C:\Users\begreen\git\3dtoolkit\Libraries\Nvpipe\Resources\paris_qcif.yuv' already exists."
```

### LINKED ISSUE #245: Setup script slow due to slow downloads of WebRTC libraries
## Report

- [x] I have searched [existing issues](https://github.com/CatalystCode/3dtoolkit/issues) and this is not a duplicate.

## Environment

- OS Version: Windows 10

- SDK Version: N/A

- Visual Studio Version: 2017

## Issues and Steps to Reproduce
***Replaces this with steps to repro your issue, if applicable.***

Downloading the WebRTC binaries from Blob Storage can be pretty slow.

## Expected Behavior
***Describe what you expected would happen.***

The .zip files are O(200MB), so they shouldn't take an unreasonable (>5 minutes) amount of time to download.

## Actual Behavior
***Describe what actually happened. Include screenshots, if applicable.***

Seeing download speeds of ~2-3Mbps. Downloads take ~15min. Slows down the install process significantly.

## Link to Code
***If you have some code that maintainers can clone/test for themselves, bugs can be resolved much faster. Please paste a link here.***

### PR DESCRIPTION
**Pull request by rozele**
## Checklist
***This checklist is used to make sure that common guidelines for a pull request are followed.***

- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [x] If applicable, the code is properly documented.
- [x] The code builds without any errors.
- [x] All unit and integration tests pass.

## Description
***Please add an informative description that covers the changes made by the pull request.***

The System.IO.Compression.ZipFile.ExtractToDirectory helper does not support overwriting existing files, so running `Libraries\Nvpipe\InstallLibraries.ps1` resulted in an exception. These updates ensure, when an updated library is made available, the previous directories are cleaned up. Other fixes include:
* Remove the need to persist the rather large .zip files.
* Ensure all install scripts are cleaning up old files before extracting new ones
* Use AzCopy if available to download dependencies.

## Related Issues and PRs
***Please provide a reference to an [existing issue](https://github.com/CatalystCode/3dtoolkit/issues) and any related pull requests.***

Fixes #237
Fixes #245

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -261,6 +261,8 @@ Libraries/glext
 *.log
 *.cso
 *.zip
+*.zip.partial
+*.zip.version
```
Comment (rozele) [inline .gitignore]: Delete.

**Comment by rozele** [comment]
@bengreenier - we could pull out some functions from DecompressZip, but the cleanup steps in each vary slightly. I think we could do this, but there's a reasonably significant testing matrix for this that makes me hesitant to any further changes. (i.e., test run setup.cmd from previous commits, test with AzCopy, test w/o AzCopy, test w/ incompatible version of AzCopy). Speaking of which, there's no unit tests for this, which is also a problem...

**Comment by bengreenier** [comment]
@rozele makes sense. in regard to the testing, yeah. :/ I always kind of felt like a package manager should own this workload anyway, so I've never invested there, but you're right if we're going to continue to use this I guess we should write some.

**Comment by anderm** [comment]
Created #264 to track unit tests for this script. Merging

**Pull request merged**